edx_sailthru/sailthru_content/services/catalog_api_service.py

```diff
@@ -30,12 +30,27 @@ def __init__(self, access_token, oauth_host, oauth_key, oauth_secret, api_url_ro
         self.api_client = EdxRestApiClient(api_url_root, jwt=self.access_token)
         self._programs_dictionary = {}
 
-    def _get_resource_from_api(self, api_endpoint, page_size, **kwargs):
-        page = 0
+    def _call_courses_api(self, **kwargs):
+        response = self.api_client.courses().get(**kwargs)
+        return response
+
+    def _call_search_api(self, **kwargs):
+        response = self.api_client.search().all().facets().get(**kwargs)
```

Comment:
If you don't need facets, please don't request them. It's an unnecessary burden on the system.